File: epicerie-vrac-projet-core/src/main/java/fr/epicerie_vrac_projet_core/models/Adresse.java
package fr.epicerie_vrac_projet_core.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@Getter
@Setter
@ToString()
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
public class Adresse {
	
	@EqualsAndHashCode.Include
	@Id
	@GeneratedValue
	private int id;
	
	private int numero;
	private String typeDeVoie;
	private String complement;
	private String codePostal;
	private String ville;
	
	@ManyToOne
	private Personne personne;
}

File: epicerie-vrac-projet-angular/src/app/models/adresse.ts
import { Personne } from "./personne";

export class Adresse {
    public id: number = 0;
    public numero: number = 0;
    public typeDeVoie: string = "";
    public complement: string = "";
    public codePostal: string = "";
    public ville: string = "";
    public personne: Personne | undefined;
}
